notify/impl.go

```diff
@@ -41,113 +40,77 @@ import (
 	"github.com/prometheus/alertmanager/types"
 )
 
-var (
-	numNotifications = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "alertmanager",
-		Name:      "notifications_total",
-		Help:      "The total number of attempted notifications.",
-	}, []string{"integration"})
-
-	numFailedNotifications = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "alertmanager",
-		Name:      "notifications_failed_total",
-		Help:      "The total number of failed notifications.",
-	}, []string{"integration"})
-)
-
-func init() {
-	prometheus.Register(numNotifications)
-	prometheus.Register(numFailedNotifications)
-}
-
 type notifierConfig interface {
 	SendResolved() bool
 }
 
-type NotifierFunc func(context.Context, ...*types.Alert) error
-
-func (f NotifierFunc) Notify(ctx context.Context, alerts ...*types.Alert) error {
-	return f(ctx, alerts...)
+// A Notifier is a type which notifies about alerts under constraints of the
+// given context.
+type Notifier interface {
+	Notify(context.Context, ...*types.Alert) error
 }
 
-type integration interface {
-	Notifier
-	name() string
+// An Integration wraps a notifier and it's config to be uniquely identified by
+// name and index from it's origin in the configuration
+type Integration struct {
+	notifier       Notifier
+	notifierConfig notifierConfig
```

Comment:
You've probably seen that the code base sticks with rather short identifier names as long as it's sufficiently clear by scope. I think `conf` is fine here and in other instances below.